metabase/metabase
# Remove redundant /user/recipient API calls

Closes #35601 

### Description

Adds a `useUserRecipient` hook that caches data so the search results + search dropdowns reuse the same data. The data is reset on refresh, so there's no issue of having stale data after a long period of time.

### How to verify
1. Type something generic into the search dropdown (like `e`)
2. Check the network tab, and see that the `/user/recipients` API call is only called once

### Checklist
